src/interface/src/app/services/notes.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '../../environments/environment';
+
+export interface Note {
+  id: number;
+  user_id: number;
+  user_name: string;
+  content: string;
+  created_at: string;
+  can_delete?: boolean;
+}
+
+export type NotesModelName = 'planning_area' | 'project_area';
+
+export interface NoteModel {
+  name: string;
+  urlMultiple: (modelObjectId: number) => string;
+  urlSingle: (modelObjectId: number, noteId: number) => string;
+}
+
+const noteEndpoints: Record<
+  NotesModelName,
+  {
+    multipleUrl: (objectId: number) => string;
+    singleUrl: (objectId: number, noteId: number) => string;
+  }
+> = {
+  planning_area: {
+    multipleUrl: (objectId: number) =>
+      `/planning/planning_area/${objectId}/note`,
+    singleUrl: (objectId: number, noteId: number) =>
+      `/planning/planning_area/${objectId}/note/${noteId}`,
+  },
+  project_area: {
+    multipleUrl: (objectId: number) =>
+      `/planning/planning_area/${objectId}/note`,
+    singleUrl: (objectId: number, noteId: number) =>
+      `/planning/planning_area/${objectId}/note/${noteId}`,
+  },
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NotesService {
```

Comment:
I'm not sure that creating a subclass here is simpler, if I'll just be injecting this in a UI component.